samples/extensions/descriptor_buffer_basic/descriptor_buffer_basic.cpp

```diff
@@ -0,0 +1,471 @@
+		buf_ptr += alignment;
+	}
+}
+
+void DescriptorBufferBasic::prepare_uniform_buffers()
+{
+	// Vertex shader scene uniform buffer block
+	uniform_buffers.scene = std::make_unique<vkb::core::Buffer>(get_device(),
+	                                                            sizeof(UboScene),
+	                                                            VK_BUFFER_USAGE_UNIFORM_BUFFER_BIT | VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT,
+	                                                            VMA_MEMORY_USAGE_CPU_TO_GPU);
+
+	// Vertex shader cube model uniform buffer blocks
+	for (auto &cube : cubes)
+	{
+		cube.uniform_buffer = std::make_unique<vkb::core::Buffer>(get_device(),
+		                                                          sizeof(glm::mat4),
+		                                                          VK_BUFFER_USAGE_UNIFORM_BUFFER_BIT | VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT,
+		                                                          VMA_MEMORY_USAGE_CPU_TO_GPU);
+	}
+
+	update_uniform_buffers();
+	update_cube_uniform_buffers(0.0f);
+}
+
+void DescriptorBufferBasic::update_uniform_buffers()
+{
+	ubo_scene.projection = camera.matrices.perspective;
+	ubo_scene.view       = camera.matrices.view;
+	uniform_buffers.scene->convert_and_update(ubo_scene);
+}
+
+void DescriptorBufferBasic::update_cube_uniform_buffers(float delta_time)
+{
+	cubes[0].model_mat = glm::translate(glm::mat4(1.0f), glm::vec3(-2.0f, 0.0f, 0.0f));
+	cubes[1].model_mat = glm::translate(glm::mat4(1.0f), glm::vec3(1.5f, 0.5f, 0.0f));
+
+	for (auto &cube : cubes)
+	{
+		cube.model_mat = glm::rotate(cube.model_mat, glm::radians(cube.rotation.x), glm::vec3(1.0f, 0.0f, 0.0f));
+		cube.model_mat = glm::rotate(cube.model_mat, glm::radians(cube.rotation.y), glm::vec3(0.0f, 1.0f, 0.0f));
+		cube.model_mat = glm::rotate(cube.model_mat, glm::radians(cube.rotation.z), glm::vec3(0.0f, 0.0f, 1.0f));
+		cube.uniform_buffer->convert_and_update(cube.model_mat);
+	}
+
+	if (animate)
+	{
+		cubes[0].rotation.x += 2.5f * delta_time;
+		if (cubes[0].rotation.x > 360.0f)
+		{
+			cubes[0].rotation.x -= 360.0f;
+		}
+		cubes[1].rotation.y += 2.0f * delta_time;
+		if (cubes[1].rotation.x > 360.0f)
+		{
+			cubes[1].rotation.x -= 360.0f;
+		}
+	}
+}
+
+void DescriptorBufferBasic::draw()
+{
+	ApiVulkanSample::prepare_frame();
+	submit_info.commandBufferCount = 1;
+	submit_info.pCommandBuffers    = &draw_cmd_buffers[current_buffer];
+	VK_CHECK(vkQueueSubmit(queue, 1, &submit_info, VK_NULL_HANDLE));
+	ApiVulkanSample::submit_frame();
+}
+
+bool DescriptorBufferBasic::prepare(vkb::Platform &platform)
+{
+	if (!ApiVulkanSample::prepare(platform))
+	{
+		return false;
+	}
+
+	/*
+	    Extension specific functions
+	*/
+
+	VkPhysicalDeviceProperties2KHR device_properties{};
+	descriptor_buffer_properties.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_DESCRIPTOR_BUFFER_PROPERTIES_EXT;
+	device_properties.sType            = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PROPERTIES_2_KHR;
```

Comment:
Slightly disturbing to set the two `sType`s here.
Maybe you should initialize `device_properties.sType` in the header?